superset/models/slice.py

```diff
@@ -140,9 +140,15 @@ def datasource_url(self) -> Optional[str]:
     def datasource_name_text(self) -> Optional[str]:
         # pylint: disable=no-member
         if self.table:
+            if self.table.schema:
+                return f"{self.table.schema}.{self.table.table_name}"
             return self.table.table_name
         datasource = self.datasource
```

Comment:
any reason why we have this here if we're not handling `self.table` the same way?

I don't really care which we do, but let's be consistent (either `table` and `datasource` or `self.table` and `self.datasource`)

Reply:
looks like most methods in this model share this inconsistency 😕